### ISSUE TITLE
Reports status passed even if the test failed

### ISSUE THREAD

**Issue by johlju** [pester/vscode-adapter#22]
When expanding the tests for file `DSC_SqlAgentAlert.Tests.ps1` then running the tests for the entire file each test (using "Run Test") fails with an error message, that is not shown when running in "Debug Test", so that is probably another issue. Though, I wasn't expecting it to report all test as passed. 🙂 

```
Build succeeded. 1 tasks, 0 errors, 0 warnings 00:00:03.3297802
C:\source\SqlServerDsc [f/support-pester-5 ≡ +0 ~3 -0 !]> . 'c:\Users\johan.ljunggren\.vscode\extensions\pspester.pester-test-0.0.2\Scripts\PesterInterface.ps1' -Discovery -PipeName PesterTestController-35192 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1 -Verbosity None
Connecting to pipe PesterTestController-35192
C:\source\SqlServerDscPS> . 'c:\Users\johan.ljunggren\.vscode\extensions\pspester.pester-test-0.0.2\Scripts\PesterInterface.ps1' -PipeName PesterTestController-35192 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:46 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:90 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:97 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:114 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:124 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:136 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:147 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:157 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:162 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:172 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:184 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:198 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:204 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:218 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:219 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:233 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:253 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:267 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:286 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:300 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:317 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:336 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:337 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:351 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:368 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:382 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:400 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:418 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:432 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:452 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:576 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:583 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:603 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:611 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:631 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:646 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:661 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:677 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:692 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:707 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:722 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:737 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:764 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:769 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:774 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:790 c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1:817
Connecting to pipe PesterTestController-35192

Starting discovery in 1 files.
Discovery finished in 36ms.
Running tests.
[-] Context DSC_SqlAgentAlert\Get-TargetResource.When Connect-SQL returns nothing failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Get-TargetResource.When the system is not in the desired state failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Get-TargetResource.When the system is in the desired state for a sql agent alert failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Test-TargetResource.When the system is not in the desired state.When the alert does not exist failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Test-TargetResource.When the system is not in the desired state.When the alert should not exist failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Test-TargetResource.When the system is not in the desired state.When enforcing a severity alert failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Test-TargetResource.When the system is in the desired state.When an alert does not exist failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Test-TargetResource.When the system is in the desired state.When enforcing a severity alert failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Test-TargetResource.When the system is in the desired state.When enforcing a message alert failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Set-TargetResource.When Connect-SQL returns nothing failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Set-TargetResource.When the system is not in the desired state and Ensure is set to Present failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

[-] Context DSC_SqlAgentAlert\Set-TargetResource.When the system is not in the desired state and Ensure is set to Absent failed
 RuntimeException: Unsupported level out output 'c:\source\SqlServerDsc\tests\Unit\DSC_SqlAgentAlert.Tests.ps1'

Tests completed in 1.5s
Tests Passed: 12, Failed: 18, Skipped: 0 NotRun: 0
BeforeAll \ AfterAll failed: 12
  - DSC_SqlAgentAlert\Get-TargetResource.When Connect-SQL returns nothing
  - DSC_SqlAgentAlert\Get-TargetResource.When the system is not in the desired state
  - DSC_SqlAgentAlert\Get-TargetResource.When the system is in the desired state for a sql agent alert
  - DSC_SqlAgentAlert\Test-TargetResource.When the system is not in the desired state.When the alert does not exist
  - DSC_SqlAgentAlert\Test-TargetResource.When the system is not in the desired state.When the alert should not exist
  - DSC_SqlAgentAlert\Test-TargetResource.When the system is not in the desired state.When enforcing a severity alert
  - DSC_SqlAgentAlert\Test-TargetResource.When the system is in the desired state.When an alert does not exist
  - DSC_SqlAgentAlert\Test-TargetResource.When the system is in the desired state.When enforcing a severity alert
  - DSC_SqlAgentAlert\Test-TargetResource.When the system is in the desired state.When enforcing a message alert
  - DSC_SqlAgentAlert\Set-TargetResource.When Connect-SQL returns nothing
  - DSC_SqlAgentAlert\Set-TargetResource.When the system is not in the desired state and Ensure is set to Present
  - DSC_SqlAgentAlert\Set-TargetResource.When the system is not in the desired state and Ensure is set to Absent
```

![image](https://user-images.githubusercontent.com/7189721/129049096-f48bb051-9d8a-4076-9ec4-91f5317acce2.png)

**Comment by JustinGrote** [comment]
I am aware of some issues where if a pester test fatally runs it won't update the status and it will maintain the previous status.

**Comment by JustinGrote** [comment]
Fixed by #34
